api/v1alpha1/oidc_types.go

```diff
@@ -62,6 +62,12 @@ type OIDC struct {
 	// If not specified, uses the default redirect URI "%REQ(x-forwarded-proto)%://%REQ(:authority)%/oauth2/callback"
 	RedirectURL *string `json:"redirectURL,omitempty"`
 
+	// Any request that matches any of the provided matchers (with either tokens that are expired or missing tokens) will not be redirected to the OIDC Provider.
+	// This behavior can be useful for AJAX or machine requests.
+	// +optional
+	// +notImplementedHide
+	OIDCDenyRedirect *OIDCDenyRedirect `json:"denyRedirect,omitempty"`
```

Comment:
Can we rename the field to `DenyRedirect`? The `OIDC` prefix feels redundant - the outer structure is already named as `OIDC`.

Reply:
agreed, this makes sense